nncf/openvino/graph/node_utils.py

```diff
@@ -280,6 +280,120 @@ def get_inplace_mean_op(reduction_axes: Optional[ReductionAxes]) -> InplaceInser
     return get_inplace_reduce_op(opset.reduce_mean, reduction_axes, False)
 
 
+def var_op(op_input: ov.Output, base_node_name: str, reduction_axes: Optional[np.ndarray] = None) -> ov.Node:
+    """
+    Return a subgraph computing variance on a given output.
+
+    :param op_input: An output to compute variance for.
+    :param base_node_name: Variance subgraph domain name.
+    :param reduction_axes: Axes along which to compute variance.
+    """
+    mean = opset.reduce_mean(
+        op_input,
+        reduction_axes=reduction_axes,
+        keep_dims=True,
+        name=f"{base_node_name}/mean",
+    )
+    diff = opset.squared_difference(mean, op_input, name=f"{base_node_name}/squared_diff")
+    variance = opset.reduce_mean(
+        diff,
+        reduction_axes=reduction_axes,
+        keep_dims=True,
+        name=f"{base_node_name}/variance",
+    )
+    return variance
+
+
+def get_inplace_mean_var_op(reduction_axes: Optional[ReductionAxes] = None) -> InplaceInsertionFnType:
+    """
+    Return an operation getter function that computes variance across given axes and then mean-reduces the result across
+    the remaining axes.
+
+    :param reduction_axes: Axes along which to compute variance.
+    """
+
+    def get_mean_var_reduce_op(node: ov.Node, output_port_id: int, output_node_name: str) -> ov.Node:
+        partial_shape = get_partial_shape_safe(node, output_port_id)
+        all_axes = np.arange(partial_shape.rank.get_length()).astype(np.int64)
+        reduction_axes_ = np.array(all_axes if reduction_axes is None else reduction_axes, dtype=np.int64)
+
+        variance = var_op(node.output(output_port_id), f"{output_node_name}/var", reduction_axes_)
+        result = opset.reduce_mean(
```

Comment:
If `ReductionAxes` is None then `opset.reduce_mean` does nothing expect `squeeze` tensor shape. I guess, the subgraph for the case when `ReductionAxes` is None can be created more optimul.

Reply:
Thanks! Done

Reply:
Reduction operations have `keep_dims` argument. Could you reuse `keep_dims`instead of explicitly using `squeeze`?

Reply:
Done